src/Core/Services/Implementations/UserService.cs

```diff
@@ -675,14 +688,13 @@ public async Task<IdentityResult> ConvertToKeyConnectorAsync(User user)
                 return IdentityResult.Failed(_identityErrorDescriber.UserAlreadyHasPassword());
             }
 
-            user.RevisionDate = user.AccountRevisionDate = DateTime.UtcNow;
-            user.MasterPassword = null;
-            user.UsesKeyConnector = true;
-
-            await _userRepository.ReplaceAsync(user);
-            await _eventService.LogUserEventAsync(user.Id, EventType.User_MigratedKeyToKeyConnector);
+            if (_currentContext.Organizations.Any(u =>
+                    u.Type is OrganizationUserType.Owner or OrganizationUserType.Admin))
+            {
+                throw new BadRequestException("Cannot use Key Connector when admin or owner of an organization.");
+            }
 
-            return IdentityResult.Success;
+            return null;
```

Comment:
It feels a little strange that this function is returning IdentityResult or null instead of a simple boolean. 
I see why it's being done though, so that the result of this method call can be returned directly in ConvertToKeyConnectorAsync.

Reply:
The alternative is to return a boolean with `var out`. Couldn't decide which was the worst option.

Reply:
Oof yeah. 
Is this method used anywhere else? Or might be used anywhere else?
All of that method's logic could be left in `ConvertToKeyConnectorAsync`.

Reply:
Both `SetKeyConnectorKeyAsync` and `ConvertToKeyConnectorAsync`.

Reply:
Nevermind, I see this is being used by `SetKeyConnectorKeyAsync` as well